streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -2268,6 +2417,32 @@ <VR, KO, VO> KTable<K, VR> leftJoin(final KTable<KO, VO> other,
                                         final TableJoined<K, KO> tableJoined,
                                         final Materialized<K, VR, KeyValueStore<Bytes, byte[]>> materialized);
 
+    /**
+     * Join records of this {@code KTable} with another {@code KTable} using non-windowed left join,
+     * using the {@link TableJoined} instance for optional configurations including
+     * {@link StreamPartitioner partitioners} when the tables being joined use non-default partitioning,
+     * and also the base name for components of the join.
+     * <p>
+     * This is a foreign key join, where the joining key is determined by the {@code foreignKeyExtractor}.
+     *
+     * @param other               the other {@code KTable} to be joined with this {@code KTable}. Keyed by KO.
+     * @param foreignKeyExtractor a {@link Function} that extracts the key (KO) from this table's value (K, V). If the
```

Comment:
This should not link to `Function` but to `BiFunction`. 

Also "from this table's value" should be "from a key-value pair in this table" I guess?

Seems like you did this for the other overloads